Sources/NIO/SocketChannel.swift

```diff
@@ -464,14 +464,17 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
         var buffer = self.recvAllocator.buffer(allocator: self.allocator)
         var readResult = ReadResult.none
 
+        // Right now we don't actually ask for any control messages. We will eventually.
+        var controlBytes = UnsafeMutableRawBufferPointer(start: nil, count: 0)
+
         for i in 1...self.maxMessagesPerRead {
             guard self.isOpen else {
                 throw ChannelError.eof
             }
             buffer.clear()
 
             let result = try buffer.withMutableWritePointer {
-                try self.socket.recvfrom(pointer: $0, storage: &rawAddress, storageLen: &rawAddressLength)
+                try self.socket.recvmsg(pointer: $0, storage: &rawAddress, storageLen: &rawAddressLength, controlBytes: &controlBytes)
```

Comment:
I think we should start migrating off the auto-magic stuff and replace `&rawAddress` and `&rawAddressLength` by `withUnsafe*`. WDYT?

Reply:
Sure, works for me.

Reply:
Oh, no, doesn't work for me. `recvmsg` is asking for inouts, not pointers, so there's no magic here at all.